test/integration/framework/utils.go

```diff
@@ -0,0 +1,309 @@
+	deploymentLabels := map[string]string{"name": "tiller"}
+	nodeSelectors := map[string]string{}
+	if len(opts.NodeSelectors) > 0 {
+		err := parseSelectorsIntoMap(opts.NodeSelectors, nodeSelectors)
+		if err != nil {
+			return nil, err
+		}
+	}
+	deployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: opts.Namespace,
+			Name:      o.TillerConfigurationOptions.DeploymentName,
+			Labels:    deploymentLabels,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: opts.getReplicas(),
+			Selector: &metav1.LabelSelector{
+				MatchLabels: deploymentLabels,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: deploymentLabels,
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:            "tiller",
+							Image:           defaultTillerImage + ":canary",
+							ImagePullPolicy: opts.pullPolicy(),
+							Ports: []corev1.ContainerPort{
+								{ContainerPort: 44134, Name: "tiller"},
+								{ContainerPort: 44135, Name: "http"},
+							},
+							Env: []corev1.EnvVar{
+								{Name: "TILLER_NAMESPACE", Value: opts.Namespace},
+							},
+							LivenessProbe: &corev1.Probe{
+								Handler: corev1.Handler{
+									HTTPGet: &corev1.HTTPGetAction{
+										Path: "/liveness",
+										Port: intstr.FromInt(44135),
+									},
+								},
+								InitialDelaySeconds: 1,
+								TimeoutSeconds:      1,
+							},
+							ReadinessProbe: &corev1.Probe{
+								Handler: corev1.Handler{
+									HTTPGet: &corev1.HTTPGetAction{
+										Path: "/readiness",
+										Port: intstr.FromInt(44135),
+									},
+								},
+								InitialDelaySeconds: 1,
+								TimeoutSeconds:      1,
+							},
+						},
+					},
+					HostNetwork:  opts.EnableHostNetwork,
+					NodeSelector: nodeSelectors,
+				},
+			},
+		},
+	}
+	return deployment, nil
+}
+
+// getFirstPodWithLabels fetches the first running pod with the desired set of labels <labelsMap>
+func (o *GardenerTestOperation) getFirstPodWithLabels(ctx context.Context, labelsMap map[string]string, namespace string, client kubernetes.Interface) (*corev1.Pod, error) {
+	var runningPodWithDesiredLabels *corev1.Pod
+
+	podList, err := getPodsByLabels(ctx, labelsMap, client, namespace)
+	if err != nil {
+		return nil, err
+	}
+
+	// Fetch the first running pod of the Kube-apiserver
+	if len(podList.Items) > 0 {
+		pod := podList.Items[0]
+		runningPodWithDesiredLabels = &pod
+	}
+	return runningPodWithDesiredLabels, nil
+}
+
+// getFirstRunningPodWithLabels fetches the first running pod with the desired set of labels <labelsMap>
+func (o *GardenerTestOperation) getFirstRunningPodWithLabels(ctx context.Context, labelsMap map[string]string, seedOrShoot ClusterType) (*corev1.Pod, error) {
+	var (
+		podList *corev1.PodList
+		err     error
+	)
+	switch seedOrShoot {
+	case Seed:
+		podList, err = getPodsByLabels(ctx, labelsMap, o.K8sSeedClient, o.Shoot.SeedNamespace)
```

Comment:
Split out the core content of this function into one that takes our `Clientset` (doesn't matter if Shoot or Seed client) and does the remaining stuff. Then, at each point where you're callin this function, instead of providing the right `ClusterType`, just use either the `K8sShootClient` or `K8sSeedClient`.

Reply:
What about this comment?